python_src/src/lamp_py/performance_manager/gtfs_utils.py

```diff
@@ -209,46 +211,30 @@ def add_parent_station_column(
     return events_dataframe
 
 
-def remove_bus_records(
-    events_dataframe: pandas.DataFrame,
-    db_manager: DatabaseManager,
-) -> pandas.DataFrame:
+def rail_routes_from_timestamp(
+    timestamp: float, db_manager: DatabaseManager
+) -> List[str]:
     """
-    remove all records from dataframe associated with bus trips
+    get a list of rail route_ids that were in effect on a given service date
+    described by a timestamp.
 
-    events_dataframe must have "static_version_key" and "route_id" columns
-
-    route_type == 3 from the routes table indicates bus service, drop all records
-    assocated with route_type == 3
+    the service_id_dates table is used to figure out the correct static version
+    key for a given service date, using the key with the max value (the keys
+    are also timestamps). then pull all the static routes with type
     """
-    process_logger = ProcessLogger(
-        "gtfs_rt.remove_bus_records",
-        start_row_count=events_dataframe.shape[0],
-    )
-    process_logger.log_start()
-
-    # unique list of "static_version_key" values for pulling route info
-    lookup_v_keys = [
-        int(s_v_key)
-        for s_v_key in events_dataframe["static_version_key"].unique()
-    ]
+    date = datetime.datetime.utcfromtimestamp(timestamp)
+    service_date = f"{date.year:04}{date.month:02}{date.day:02}"
 
-    # pull non-bus route_id's from RDS
-    non_bus_id_query = sa.select(
-        StaticRoutes.static_version_key,
-        StaticRoutes.route_id,
-    ).where(
-        StaticRoutes.static_version_key.in_(lookup_v_keys),
-        StaticRoutes.route_type != 3,
+    svk_subquery = (
+        sa.select(sa.func.max(ServiceIdDates.static_version_key))
+        .where(ServiceIdDates.service_date == service_date)
+        .scalar_subquery()
     )
-    non_bus_ids = db_manager.select_as_dataframe(non_bus_id_query)
-
-    # join events on non-bus "route_id"s and "static_version_key" foreign key
-    events_dataframe = events_dataframe.merge(
-        non_bus_ids, how="inner", on=["static_version_key", "route_id"]
+    result = db_manager.execute(
+        sa.select(StaticRoutes.route_id).where(
+            StaticRoutes.route_type.in_([0, 1, 2]),
+            StaticRoutes.static_version_key == svk_subquery,
+        )
     )
```

Comment:
nitpick: Since we're already doing a very similar lookup inside of `add_static_version_key_column` we could refactor `add_static_version_key_column` a bit, moving the RDS lookup logic into a separate function and calling that here and in `add_static_version_key_column` for uniformity. 

`def get_static_version_key_from_service_date(service_date: int, db_manager: DatabaseManager) -> int:`

Reply:
do you prefer the logic here with `ServiceIdDates` and `max` or the logic in `add_static_version_key_column`? this logic is certainly simpler, as SA code, but its dependent on a view that might take more time to execute?

Reply:
I prefer the `add_static_version_key_column` logic. It's more accurate for what we're trying to do there and should execute faster.